cmd/security-file-token-provider/main.go

```diff
@@ -0,0 +1,123 @@
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+// Constants
+
+// Dependencies
+
+var fileProvider fileprovider.TokenProvider
+
+func main() {
+	startupTimer := startup.NewStartUpTimer(1, internal.BootTimeoutDefault)
+
+	var useRegistry bool
+	var profileDir, configDir string
+
+	flag.BoolVar(&useRegistry, "registry", false, "Indicates the service should use registry service.")
+	flag.BoolVar(&useRegistry, "r", false, "Indicates the service should use registry service.")
+	flag.StringVar(&profileDir, "profile", "", "Specify a profile other than default.")
+	flag.StringVar(&profileDir, "p", "", "Specify a profile other than default.")
+	flag.StringVar(&configDir, "confdir", "", "Specify local configuration directory")
+
+	flag.Usage = usage.HelpCallback
+	flag.Parse()
+
+	bootstrap.Run(
+		configDir,
+		profileDir,
+		internal.ConfigFileName,
+		useRegistry,
+		clients.SupportLoggingServiceKey,
+		fileprovider.Configuration,
+		startupTimer,
+		[]bootstrapinterface.BootstrapHandler{
+			func(
+				wg *sync.WaitGroup,
+				ctx context.Context,
+				startupTimer startup.Timer,
+				config bootstrapinterface.Configuration,
+				logging logger.LoggingClient,
+				registry registry.Client) bool {
+
+				return runTokenProvider(logging)
+			},
+		})
+}
+
+func runTokenProvider(logging logger.LoggingClient) bool {
+
+	cfg := fileprovider.Configuration
+
+	fileOpener := fileioperformer.NewDefaultFileIoPerformer()
+	tokenProvider := authtokenreader.NewAuthTokenReader(fileOpener)
+	tokenConfigParser := tokenconfig.NewTokenConfigParser(fileOpener)
+
+	var req internal.HttpCaller
+	if caFilePath := cfg.SecretService.CaFilePath; caFilePath != "" {
+		logging.Info("using certificate verification for secret store connection")
+		caReader, err := fileOpener.OpenFileReader(caFilePath, os.O_RDONLY, 0400)
+		if err != nil {
+			logging.Error(fmt.Sprintf("failed to load CA certificate: %s", err.Error()))
+			return false
+		}
+		req = vaultclient.NewRequestor(logging).WithTLS(caReader, cfg.SecretService.ServerName)
+	} else {
+		logging.Info("bypassing certificate verification for secret store connection")
+		req = vaultclient.NewRequestor(logging).Insecure()
+	}
+	vaultScheme := cfg.SecretService.Scheme
+	vaultHost := fmt.Sprintf("%s:%v", cfg.SecretService.Server, cfg.SecretService.Port)
+	vaultClient := vaultclient.NewVaultClient(logging, req, vaultScheme, vaultHost)
+
+	if fileProvider == nil {
+		// main_test.go has injected a testing mock if fileProvider != nil
+		fileProvider = fileprovider.NewTokenProvider(logging, fileOpener, tokenProvider, tokenConfigParser, vaultClient)
+	}
+
+	fileProvider.SetConfiguration(cfg.SecretService, cfg.TokenFileProvider)
+	err := fileProvider.Run()
+
+	if err != nil {
+		logging.Error(fmt.Sprintf("error occurred generating tokens: %s", err.Error()))
+	}
+
+	return false // Tell bootstrap.Run() to exit wait loop and terminate
```

Comment:
Nitpick - implementation detail required to fulfill bootstrapHandler contract.  I'd remove this from here (see above).